src/kafka-handler.ts

```diff
@@ -70,57 +93,100 @@ class KafkaHandler implements DataBroker {
         // First try
         // Creating consumer client - from device manager to iotagent.
         // This is always used.
-        let client = new kafka.Client(config.device_manager.kafkaHost, "iotagent-json-consumer-" +  Math.floor(Math.random() * 10000));
-        console.log("... Kafka client for consumer is ready.");
         console.log("Creating Kafka consumer...");
-        this.consumer = new kafka.HighLevelConsumer(client, config.device_manager.kafkaTopics, config.device_manager.kafkaOptions);
-        // Kafka consumer events registration
-        this.consumer.on("message", (data) => {
-          let parsedData = JSON.parse(data.value.toString());
-          callback(parsedData);
-        });
-        this.consumer.on("error", (err) => {
-          if (isScheduled == true) {
-            console.log("An operation was already scheduled. No need to do it again.");
-            return;
+        this.getPublishTopic('admin', config.device_manager.inputSubject, (err?:any, topic?: string) => {
```

Comment:
Is 'admin' also used for all devices? If I got it right, DeviceManager will use the device tenant to get the topic to which all data is published.

Reply:
No, 'admin' should be used only for the 'admin' tenant.
Commit 12052ef fixes this.